xbmc/windowing/gbm/DRMUtils.cpp

```diff
@@ -128,6 +128,80 @@ drm_fb * CDRMUtils::DrmFbGetFromBo(struct gbm_bo *bo)
   return fb;
 }
 
+static bool GetProperties(int fd, uint32_t id, uint32_t type, struct drm_object *object)
+{
+  drmModeObjectPropertiesPtr props = drmModeObjectGetProperties(fd, id, type);
+  if (!props)
+    return false;
+
+  object->id = id;
+  object->type = type;
+  object->props = props;
+  object->props_info = new drmModePropertyPtr[props->count_props];
+
+  for (uint32_t i = 0; i < props->count_props; i++)
+    object->props_info[i] = drmModeGetProperty(fd, props->props[i]);
+
+  return true;
+}
+
+static void FreeProperties(struct drm_object *object)
+{
+  if (object->props_info)
+  {
+    for (uint32_t i = 0; i < object->props->count_props; i++)
+      drmModeFreeProperty(object->props_info[i]);
+
+    delete [] object->props_info;
+    object->props_info = nullptr;
+  }
+
+  drmModeFreeObjectProperties(object->props);
+  object->props = nullptr;
+  object->type = 0;
+  object->id = 0;
+}
+
+static uint32_t GetPropertyId(struct drm_object *object, const char *name)
+{
+  for (uint32_t i = 0; i < object->props->count_props; i++)
+    if (!strcmp(object->props_info[i]->name, name))
+      return object->props_info[i]->prop_id;
+
+  CLog::Log(LOGWARNING, "CDRMUtils::%s - could not find property %s", __FUNCTION__, name);
+  return 0;
+}
+
+bool CDRMUtils::AddProperty(drmModeAtomicReqPtr req, struct drm_object *object, const char *name, uint64_t value)
+{
+  uint32_t property_id = GetPropertyId(object, name);
+  if (!property_id)
+    return false;
+
+  if (drmModeAtomicAddProperty(req, object->id, property_id, value) < 0)
+  {
+    CLog::Log(LOGERROR, "CDRMUtils::%s - could not add property %s", __FUNCTION__, name);
+    return false;
+  }
+
+  return true;
+}
+
+bool CDRMUtils::SetProperty(struct drm_object *object, const char *name, uint64_t value)
+{
+  uint32_t property_id = GetPropertyId(object, name);
+  if (!property_id)
+    return false;
+
+  if (drmModeObjectSetProperty(m_fd, object->id, object->type, property_id, value))
```

Comment:
was this supposed to test < 0 ?

Reply:
looking at modetest code it treats `< 0` as error but the ioctl call in `drmModeObjectSetProperty` should return 0 on success

Reply:
I will include a fix in an upcoming limit gui size PR